pkg/kubelet/cm/devicemanager/topology_hints.go

```diff
@@ -136,7 +136,7 @@ func (m *ManagerImpl) GetPodTopologyHints(pod *v1.Pod) map[string][]topologymana
 	return deviceHints
 }
 
-func (m *ManagerImpl) deviceHasTopologyAlignment(resource string) bool {
+func (m *ManagerImpl) deviceHasTopologyAlignmentLocked(resource string) bool {
```

Comment:
Can you explain this renaming? I don't see any locks in the code.

Reply:
@huyinhou ^^^^

Reply:
Instead of introducing this extra function, I think it would be cleaner to turn `m.mutex` into a RWMutex and think more carefully about who is a reader vs. writer when locking.

Reply:
`devicesToAllocate` calls `deviceHasTopologyAlignment` with mutex locked, but `GetTopologyHints` and `GetPodTopologyHints` call it without mutex locked, so we have to add a new function to handle these two situations。  
RWMutex is a good choice, I think we can refactor the code structure and turn mutex into RWMutex in another PR.

Reply:
I guess I'm hung-up on why we would put the lock at such a low-level within `deviceHasTopologyAlignment` in the first place.

If there's a race between a plugin changing the set of devices it has registered and generating topology hints for those devices, then I would think we want to lock the entire topology hint generation function, and not just this low level function of checking if an individual device has topology alignment or not.